Release - dependency updates in release notes

OscarSpruit opened the pull request:
## Description
Release notes for dependency updates will now be automatically generated by comparing the `.toml` files from previous release and the current.

Example output:

| Name | Version |
|------|---------|
| [Adyen 3DS2](https://github.com/Adyen/adyen-3ds2-android/releases) | `2.2.18` -> `2.2.21` |
| [AndroidX Fragment](https://developer.android.com/jetpack/androidx/releases/fragment) | `1.8.0` -> `1.8.4` |
| [AndroidX Lifecycle Runtime](https://developer.android.com/jetpack/androidx/releases/lifecycle) | `2.8.2` -> `2.8.3` |
| [AndroidX Lifecycle ViewModel](https://developer.android.com/jetpack/androidx/releases/lifecycle) | `2.8.2` -> `2.8.3` |
| [AndroidX RecyclerView](https://developer.android.com/jetpack/androidx/releases/recyclerview) | `1.2.2` -> `1.3.2` |
| [CashApp Pay](https://github.com/cashapp/cash-app-pay-android-sdk/releases) | `2.4.0` -> `2.5.0` |
| [AndroidX Activity Compose](https://developer.android.com/jetpack/androidx/releases/activity) | `1.9.1` -> `1.9.2` |
| [AndroidX Lifecycle ViewModel Compose](https://developer.android.com/jetpack/androidx/releases/lifecycle) | `2.8.2` -> `2.8.3` |
| [AndroidX Compose Compiler](https://developer.android.com/jetpack/androidx/releases/compose-compiler) | `1.5.14` -> `1.5.15` |
| [Play Services Wallet](https://developers.google.com/pay/api/android/support/release-notes) | `19.3.0` -> `19.4.0` |
| [Material](https://github.com/material-components/material-components-android/releases) | `1.11.0` -> `1.12.0` |
| [OkHttp](https://square.github.io/okhttp/changelogs/changelog/) | `4.11.0` -> `4.12.0` |
| [WeChat](https://developers.weixin.qq.com/doc/oplatform/en/Mobile_App/Access_Guide/Android.html) | `6.7.0` -> `6.8.0` |
| Twint | `7.0.0` -> `8.0.0` |
| [Android Gradle Plugin](https://developer.android.com/build/releases/gradle-plugin) | `8.6.0` -> `8.7.1` |

## Checklist <!-- Remove any line that's not applicable -->
- [x] PR is labelled <!-- Breaking change, Feature, Fix, Dependencies or Chore -->
- [x] Changes are tested manually

COAND-1017

jreij commented on .github/.release_notes_dependency_exclusion_list:
```diff
@@ -0,0 +1,54 @@
+com.google.dagger:hilt-android-testing
+com.google.dagger:hilt-compiler
+com.google.devtools.ksp
+com.google.pay.button:compose-pay-button
```
Will this be part of the SDK with Google Pay improvements PR?

OscarSpruit replied:
Yes, so depending on which PR we move first, we should remove it either in this PR or in the Google Pay one

jreij replied:
I would say remove it from here regardless so that we don't forget it

jreij commented on scripts/generate_release_notes.py:
```diff
@@ -0,0 +1,204 @@
+    return commit[start:end]
+
+def fetch_pr(number: str) -> dict:
+    url = 'https://api.github.com/repos/{}/pulls/{}'.format(os.getenv('GITHUB_REPO'), number)
```
What happens if we update a dependency without using renovate? Will this script skip it?

OscarSpruit commented on .github/workflows/publish_release.yml:
```diff
@@ -5,23 +5,9 @@ on:
-    needs: [generate_version_name, publish_to_maven_central]
-    with:
-      version-name: ${{ needs.generate_version_name.outputs.version-name }}
+      version-name: "9.9.9"
```
No, this is for testing. I'll revert that commit later

araratthehero commented:
@OscarSpruit I tested the new script and noticed that the newlines between release notes from separate PRs are not being removed anymore.

PR1:
```
### Fixed
- Fixed some important bug
```

PR2:
```
### Fixed
- Fixed something
- Fixed something else
```

Expected result:
```
### Fixed
- Fixed some important bug
- Fixed something
- Fixed something else

```

Actual result:
```
### Fixed
- Fixed some important bug

- Fixed something
- Fixed something else

```

I also noticed that there are double empty lines between `Improved` and `Changed` headers:
```
### Improved
- Made improvements in multiple modules


### Changed
- For Drop-in, headers of preselected stored payment screen and payment methods list screen are updated.
```

Tested it [here](https://github.com/Adyen/adyen-testcards-android/actions/runs/12232817677).

OscarSpruit commented:
@araratthehero thanks for double checking! I fixed the script to format everything properly. The output is like this now:

### New
- You can now use [Adyen Test Cards Android](https://github.com/Adyen/adyen-testcards-android) to autofill test payment method information, to test your integration more quickly.
> [!WARNING]
> Some warning
- A list
  - List item 1
  - List item 2
- Table

  | Class                       | Description                        |
  |-----------------------------|------------------------------------|
  | `CardNumberValidator`       | Validates the card number field.   |
  | `CardExpiryDateValidator`   | Validates the expiry date field.   |
  | `CardSecurityCodeValidator` | Validates the security code field. |
- Added the ability to copy payment method data to the clipboard

### Fixed
- Fixed some important bug
- Fixed something
- Fixed something else

### Improved
- Made improvements in multiple modules

### Changed
- For Drop-in, headers of preselected stored payment screen and payment methods list screen are updated.
- When you set `analyticsConfiguration = AnalyticsConfiguration(AnalyticsLevel.NONE)`, only [Drop-in/Components analytics](https://docs.adyen.com/online-payments/analytics-and-data-tracking#data-we-are-collecting) are not sent to Adyen.
- List with links
  | Name                                                                                                   | Version                       |
  |--------------------------------------------------------------------------------------------------------|-------------------------------|
  | [Android Gradle Plugin](https://developer.android.com/build/releases/past-releases/agp-8-5-0-release-notes#android-gradle-plugin-8.5.1)                          | **8.5.1**                    |
  | [AndroidX Fragment](https://developer.android.com/jetpack/androidx/releases/fragment#1.8.3)            | **1.8.3**                     |
  | [AndroidX Activity](https://developer.android.com/jetpack/androidx/releases/activity#1.9.2)            | **1.9.2**                     |
  | [AndroidX Compose Activity](https://developer.android.com/jetpack/androidx/releases/activity#1.9.2)    | **1.9.2**                     |
  | [AndroidX Compose BOM](https://developer.android.com/develop/ui/compose/bom/bom-mapping)               | **2024.06.00**                |
  | [AndroidX Lifecycle](https://developer.android.com/jetpack/androidx/releases/lifecycle#2.8.3)          | **2.8.3**                     |
  | [AndroidX Lifecycle ViewModel Compose](https://developer.android.com/jetpack/androidx/releases/lifecycle#2.8.3)  | **2.8.3**                     |
  | [AndroidX AppCompat](https://developer.android.com/jetpack/androidx/releases/appcompat#1.7.0)          | **1.7.0**                     |

- Dependency versions:
  | Name | Version |
  |------|---------|
  | [AndroidX Activity Compose](https://developer.android.com/jetpack/androidx/releases/activity) | `1.9.2` -> `1.9.3` |
  | [AndroidX Compose BoM](https://developer.android.com/develop/ui/compose/bom/bom-mapping) | `2024.09.02` -> `2024.10.01` |
